editor/src/clj/dynamo/graph.clj

```diff
@@ -1582,19 +1582,29 @@
      true)))
 
 (defn node-property-dynamic
-  "Returns the value of a dynamic associated with a specific node property."
-  [node property-label dynamic-label evaluation-context]
-  (let [node-type (node-type node)
-        property-def (get (in/all-properties node-type) property-label)
-        _ (assert property-def (format "Property %s not found in node-type %s." property-label (:k node-type)))
-        dynamic-fnk (-> property-def (get :dynamics) (get dynamic-label))
-        _ (assert dynamic-fnk (format "Dynamic %s not found on property %s in node-type %s." dynamic-label property-label (:k node-type)))
-        dynamic-fn (:fn dynamic-fnk)
-        dynamic-args (coll/transfer (:arguments dynamic-fnk) {}
-                       (map (fn [arg-label]
-                              (let [arg-value (in/node-value node arg-label evaluation-context)]
-                                (pair arg-label arg-value)))))]
-    (dynamic-fn dynamic-args)))
+  "Returns the value of a dynamic associated with a specific node property
+  If the dynamic does not exist, returns the provided not-found value, or throws
+  an assertion error if there was no not-found value provided"
+  ([node property-label dynamic-label evaluation-context]
+   (let [ret (node-property-dynamic node property-label dynamic-label ::not-found evaluation-context)]
+     (assert (not (identical? ret ::not-found))
+             (format "Dynamic %s not found on property %s in node-type %s." dynamic-label property-label (:k (node-type node))))
+     ret))
```

Comment:
When I started providing defaults in the places that didn't do that explicitly, I realized they actually expected the dynamic to be defined. So instead made this function throw if `not-found` was not explicitly provided when there is no dynamic.